realm/realm-library/src/androidTest/kotlin/io/realm/KotlinSchemaTests.kt

```diff
@@ -49,7 +49,7 @@ class KotlinSchemaTests {
 
     @Test
     fun kotlinTypeNonNull() {
-        val objSchema = realm.schema.get(AllKotlinTypes::class.simpleName)!!
+        val objSchema = realm.schema.get(AllKotlinTypes::class.simpleName!!)!!
 
         // Document current nullability. Ideally all should be non-nullable. This is currently
```

Comment:
Yes exactly. This means that we should add support for `@Nonnull` as an alternative way of defining required types. This would make the interop be seamless with Kotlin as this is what they do in their bytecode. Of course, it also means that you could write: `
@NonNull public String name;` instead of `@Required public String name` in Java, but that is IMO not a big problem. We shouldn't encourage it either though.